### PR TITLE
Keyboard accessibility improvements for search form

### PR DESCRIPTION
**Pull request by paulgibbs**
Per https://github.com/squidfunk/mkdocs-material/issues/400#issuecomment-316968253 this change:

* Removes `accesskey`. This apparently is not recommend and accessibiltiy/browser implementation is too inconsistent.
* I've also removed the JS-powered "F" and "S" key binds that did basically the same thing. Same reason as above, but also because these were an internationalisation issue ("find" and "search" will probably not not begin with those letters in other languages).
* I've added the `dialog` role to the search model. There's more to do here (we need to add `aria-labelledby`), but that can be addressed in the future when the triple `label` elements are rationalised. I don't think adding the ARIA role hurts in the meanwhile.
* The tabbing JS was really good but I've made two improvements.
  * There was inconsistent modal behaviour if you hit tab and you had not made a search yet (i.e. no results).
  * Tab now cycles back to the input element, allowing keyboard users the ability to refine their search terms.

In testing, this makes the search tabbing feel much better to me.

I don't know modern JS very well, so I've tried my best to keep to your code standards, but do let me know if I should have done something different there so I can learn. Thanks!

### CODE REVIEW

**Comment by paulgibbs** [comment]
This would also address #398 :)

**Comment by squidfunk** [comment]
Thank you for this PR. It includes a lot of changes that need to be tested and validated cross-browser. As there are currently not end-to-end tests, this may take some time.

**Comment by squidfunk** [comment]
I'm sorry but I will not completely merge this PR, because I personally find the <kbd>S</kbd> and <kbd>F</kbd> short cuts really practical. GitHub also has a <kbd>S</kbd> shortcut defined, so it's pretty common. I will incorporate the other changes you proposed in separate PRs (this PR is too big).

File: src/assets/javascripts/application.js [hunk 1]
```diff
@@ -302,9 +302,10 @@ function initialize(config) { // eslint-disable-line func-style
 
         /* Retrieve all results */
         const links = Array.prototype.slice.call(
-          document.querySelectorAll("[data-md-component=search] [href]"))
-        if (!links.length)
-          return
+          document.querySelectorAll(
+            ".md-search__input, [data-md-component=search] [href]"
```
Comment (squidfunk) [inline src/assets/javascripts/application.js]: Also, why did you change this?

Comment (paulgibbs) [inline src/assets/javascripts/application.js]: The removal: makes tab behaviour consistent before you have made any search.
e.g. on master, load your site, hit tab several times. Then load your site again, make a search (so results show up), then hit tab several times. There is a behaviour difference and the removal was part of making the behaviour consistent.

The addition: lets you tab back to the search input box (otherwise how does a keyboard user go back to that box to refine their search?)

**Comment by paulgibbs** [comment]
No worries about not merging the whole thing. Even if you can't cherry-pick specific commits, hopefully having the code will help you make improvements, anyway.

**Comment by squidfunk** [comment]
Yes, thank you for investing your time, I'm about to incorporate your improvements with minor changes!

File: src/assets/javascripts/application.js (hunk 1 above)
Comment (squidfunk) [inline src/assets/javascripts/application.js]: This is actually an awesome idea, feels much more natural.

**Comment by squidfunk** [comment]
> Removes `accesskey`. This apparently is not recommend and accessibiltiy/browser implementation is too inconsistent.

Implemented in #406

> I've also removed the JS-powered "F" and "S" key binds that did basically the same thing. Same reason as above, but also because these were an internationalisation issue ("find" and "search" will probably not not begin with those letters in other languages).

Not implemented for reasons stated above

> I've added the `dialog` role to the search model. There's more to do here (we need to add `aria-labelledby`), but that can be addressed in the future when the triple `label` elements are rationalised. I don't think adding the ARIA role hurts in the meanwhile.

Implemented in #406 

> The tabbing JS was really good but I've made two improvements.
>  * There was inconsistent modal behaviour if you hit tab and you had not made a search yet (i.e. no results).
>  * Tab now cycles back to the input element, allowing keyboard users the ability to refine their search terms.

Implemented in #410
